packages/core/src/utils/CollectLabels.ts

```diff
@@ -285,8 +294,42 @@ const setPendingProperty = (
   }
 };
 
+const insertPendingHelper = (
+  shapes: ShapeAD[],
+  xs: number[],
+  state: State
+): void => {
+  for (const { shapeType, properties } of shapes) {
+    if (shapeType === "Group") {
+      const subShapes = getValueAsShapeList(properties["shapes"]);
+      insertPendingHelper(subShapes, xs, state);
+    } else {
+      const shapedef: ShapeDef = shapedefs[shapeType];
+      if (properties.name.tag === "StrV") {
+        const labelData = state.labelCache.get(properties.name.contents);
+        if (labelData !== undefined) {
+          for (const propertyID of shapedef.pendingProps) {
+            setPendingProperty(
+              properties,
+              propertyID,
+              labelData[propertyID],
+              xs,
+              state.inputs
+            );
+          }
+        }
+      }
+    }
+  }
+};
+
 export const insertPending = (state: State): State => {
   const varyingValues = [...state.varyingValues];
+  insertPendingHelper(
+    state.shapes,
+    varyingValues,
+    state
+  ); /*
```

Comment:
Could you delete this code instead of commenting it out?

Reply:
Done!